File: lc451.cs
public class Solution {
    public string FrequencySort(string s) {
        Dictionary<char,int> map = new Dictionary<char,int>();
        foreach(char c in s)
        {
            if(!map.ContainsKey(c))
            {
                map.Add(c,0);
            }
            map[c]++;
        }
        StringBuilder sb= new StringBuilder();
        foreach(var kvp in map.OrderByDescending(m=>m.Value))
        {
            sb.Append(new string(kvp.Key,kvp.Value));
        }
        return sb.ToString();
    }
}

File: java/lc451.java
class Solution {
    public String frequencySort(String s) {      
        Map<Character, Integer> unSortedMap = new HashMap();
        LinkedHashMap<Character, Integer> reverseSortedMap = new LinkedHashMap<>();
        for(Character c : s.toCharArray())
        {
            if(!unSortedMap.containsKey(c))
            {
                unSortedMap.put(c,1);
            }
            else
            {
                unSortedMap.put(c, 1+unSortedMap.get(c));
            }
        }
        unSortedMap.entrySet()
            .stream()
            .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder())) 
            .forEachOrdered(x -> reverseSortedMap.put(x.getKey(), x.getValue()));
        StringBuilder sb= new StringBuilder();

        Set<Character> keys = reverseSortedMap.keySet();
        for(Character key: keys){
            sb.append(key.toString().repeat(reverseSortedMap.get(key)));
        }
        return sb.toString();
    
    }
}
